airflow/models/baseoperator.py

```diff
@@ -362,6 +362,9 @@ def partial(
     partial_kwargs["end_date"] = timezone.convert_to_utc(partial_kwargs["end_date"])
     if partial_kwargs["pool"] is None:
         partial_kwargs["pool"] = Pool.DEFAULT_POOL_NAME
+    if partial_kwargs["pool_slots"] < 1:
+        dag_str = f" in dag {partial_kwargs['dag'].dag_id}" if partial_kwargs["dag"] else ""
```

Comment:
Should this use `.get()` like in the decorator case? It might be worthwhile to split this line into two for readability.

Reply:
I used .get() only in the _expand() method because I got an error during pre-commit without it. Additionally, all conditional statements in _expand() used .get() whereas none of the conditional statements in partial() used .get(). I believe this might be because the partial_kwargs dictionary is initialized at the beginning of the partial() method with all key: value pairs including "pool_slots" (see line 319 of baseoperator.py) to prevent a KeyError, but this complete initialization of all key: value pairs doesn't happen in _expand() leading to potential KeyErrors, which is why .get() must be used in this case. If you still think I should add .get() to partial() just to be safe, I can do it. I just wanted to check with you because it might not be uniform with other conditional statements in partial(). Please let me know!

I will also go ahead and split the f strings into two lines in the partial() and _expand() methods.